src/Stack/Fetch.hs

```diff
@@ -22,57 +22,59 @@ module Stack.Fetch
+import           Control.Concurrent.MVar.Lifted (modifyMVar, newMVar)
+import           Control.Concurrent.STM         (TVar, atomically, modifyTVar,
+                                                 newTVarIO, readTVar,
+                                                 readTVarIO, writeTVar)
+import           Control.Exception              (assert)
+import           Control.Monad                  (join, liftM, unless, void,
+                                                 when)
 import           Control.Monad.Catch
 import           Control.Monad.IO.Class
 import           Control.Monad.Logger
-import           Control.Monad.Reader (asks)
-import           Control.Monad.Reader (runReaderT)
+import           Control.Monad.Reader           (asks)
+import           Control.Monad.Reader           (runReaderT)
 import           Control.Monad.Trans.Control
-import           Crypto.Hash (SHA512(..))
-import           Data.ByteString (ByteString)
-import qualified Data.ByteString as S
-import qualified Data.ByteString.Lazy as L
-import           Data.Either (partitionEithers)
-import qualified Data.Foldable as F
-import           Data.Function (fix)
-import           Data.IORef (newIORef, readIORef, writeIORef)
-import           Data.List (intercalate)
-import           Data.Map (Map)
-import qualified Data.Map as Map
-import           Data.Maybe (maybeToList)
-import           Data.Monoid ((<>))
-import           Data.Set (Set)
-import qualified Data.Set as Set
-import qualified Data.Text as T
-import           Data.Text.Encoding (decodeUtf8)
-import           Data.Typeable (Typeable)
-import           Data.Word (Word64)
+import           Crypto.Hash                    (SHA512 (..))
+import           Data.ByteString                (ByteString)
+import qualified Data.ByteString                as S
+import qualified Data.ByteString.Lazy           as L
+import           Data.Either                    (partitionEithers)
+import qualified Data.Foldable                  as F
+import           Data.Function                  (fix)
+import           Data.IORef                     (newIORef, readIORef,
+                                                 writeIORef)
+import           Data.List                      (intercalate, intersperse)
+import           Data.Map                       (Map)
+import qualified Data.Map                       as Map
+import           Data.Maybe                     (maybeToList)
+import           Data.Monoid                    ((<>))
+import           Data.Set                       (Set)
+import qualified Data.Set                       as Set
+import qualified Data.Text                      as T
+import           Data.Text.Encoding             (decodeUtf8)
+import           Data.Typeable                  (Typeable)
+import           Data.Word                      (Word64)
 import           Network.HTTP.Download
 import           Path
-import           Prelude -- Fix AMP warning
+import           Prelude
```

Comment:
This comment shouldn't get lost, as someone may then accidentally remove the import.